p2p/host/autorelay/options.go

```diff
@@ -10,9 +10,22 @@ import (
 	"github.com/benbjohnson/clock"
 )
 
+// AutoRelay will call this function when it needs new candidates because it is
+// not connected to the desired number of relays or we get disconnected from one
+// of the relays. Implementations must send *at most* numPeers, and close the
+// channel when they don't intend to provide any more peers. AutoRelay will not
+// call the callback again until the channel is closed. Implementations should
+// send new peers, but may send peers they sent before. AutoRelay implements a
+// per-peer backoff (see WithBackoff). See WithMinInterval for setting the
+// minimum interval between calls to the callback. The context.Context passed
+// MAY be canceled when AutoRelay feels satisfied, it will be canceled when the
+// node is shutting down. If the channel is canceled you MUST close the output
+// channel at some point.
+type PeerSource func(ctx context.Context, num int) <-chan peer.AddrInfo
```

Comment:
I don't think this is correct. We're passing in a `context.Context` here, not the `context.CancelFunc`, so AutoRelay has no way of canceling this context.

Reply:
AutoRelay is the one passing in the context. There is a typo in this comment. It should say If the context is canceled instead of if the channel is canceled.

Reply:
Right, sorry for the stupid mistake.